src/firefly/java/edu/caltech/ipac/firefly/util/ImageSetConverter.java

```diff
@@ -0,0 +1,188 @@
+package edu.caltech.ipac.firefly.util;
+
+import edu.caltech.ipac.firefly.server.util.DsvToDataGroup;
+import edu.caltech.ipac.firefly.server.util.Logger;
+import edu.caltech.ipac.firefly.server.visualize.imagesources.ImageMasterDataEntry;
+import edu.caltech.ipac.util.DataGroup;
+import edu.caltech.ipac.util.DataObject;
+import edu.caltech.ipac.util.DataType;
+import edu.caltech.ipac.util.download.FailedRequestException;
+import org.apache.commons.csv.CSVFormat;
+import org.json.simple.JSONObject;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Created by zhang on 9/27/17.
+ * This class will read in the ImageSet information table (maybe from an API) and
+ * the convert to a JSONObject.
+ */
+public class ImageSetConverter {
+    private  final String masterTablePath = "/edu/caltech/ipac/firefly/resources/";
+    private  final String irsaMasterTableName = "irsa-master-table.csv";
+    private  final String lsstMasterTableName = "lsst-master-table.csv";
+    private static final Logger.LoggerImpl LOG = Logger.getLogger();
+
+    private static HashMap<String, String> paramMaps = new HashMap<>();
+    List<ImageMasterDataEntry>  retList= new ArrayList();
+
+    public ImageSetConverter(String whichProject) throws IOException, FailedRequestException {
+
+        //define the key mappings between the master table to the json file
+        paramMaps.put("apiType", "Service");
+        paramMaps.put("dataProductId", "surveyKey");
+        paramMaps.put("waveBandId", "surveyKeyBand");
+        paramMaps.put("waveBandDesc", "title");
+        paramMaps.put("filter", "filter");
+
+        String masterTable=null;
+        switch (whichProject.toLowerCase()){
+            case "irsa":
+                masterTable= new String(masterTablePath+irsaMasterTableName);
+                break;
+            case "lsst":
+                masterTable= new String(masterTablePath+lsstMasterTableName);
+                break;
+        }
+        retList=createDataList(masterTable);
+
+    }
+
+    /**
+     * This method reads in the master-table and then create a Map List based on the table data.
+     * @return
+     * @throws IOException
+     * @throws FailedRequestException
+     */
+    public List createDataList(String masterTable) throws IOException, FailedRequestException {
+
+        DataGroup inDg = getDataFromMasterTable(masterTable);
+        DataType[] dataTypes = inDg.getDataDefinitions();
+        List<DataObject> dataRows = inDg.values();
+
+
+        for(DataObject obj : dataRows) {
+            Map<String,Object> map= new HashMap<>();
+            map.putAll(dataObjectToMap(obj,dataTypes ));
+            retList.add( ImageMasterDataEntry.makeFromMap(map));
+        }
```

Comment:
This method is not made to be used like this.  Setters should be called or should have a list of constants and a set with each constants.  Otherwise we are going to put junk in the object.

Reply:
Yes, we already covered that yesterday.